org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/AdvancedQuickAssistProcessor.java

```diff
@@ -1982,154 +1999,183 @@ private static Expression combineOperands(ASTRewrite rewrite, Expression existin
-	//					Object o = iter.next();
-	//					if (!renamedNames.contains(o)) {
-	//						return false;
-	//					}
-	//				}
-	//				// check operator and replace if needed
-	//				Assignment.Operator operator = assignment.getOperator();
-	//				if (operator == Assignment.Operator.BIT_AND_ASSIGN) {
-	//					Assignment newAssignment = ast.newAssignment();
-	//					newAssignment.setLeftHandSide(newName);
-	//					newAssignment.setRightHandSide(inversedExpression);
-	//					newAssignment.setOperator(Assignment.Operator.BIT_OR_ASSIGN);
-	//					rewrite.replace(assignment, newAssignment, null);
-	//				} else if (operator == Assignment.Operator.BIT_OR_ASSIGN) {
-	//					Assignment newAssignment = ast.newAssignment();
-	//					newAssignment.setLeftHandSide(newName);
-	//					newAssignment.setRightHandSide(inversedExpression);
-	//					newAssignment.setOperator(Assignment.Operator.BIT_AND_ASSIGN);
-	//					rewrite.replace(assignment, newAssignment, null);
-	//				} else {
-	//					rewrite.replace(expression, inversedExpression, null);
-	//					// set new name
-	//					rewrite.replace(name, newName, null);
-	//				}
-	//			} else if (location == VariableDeclarationFragment.NAME_PROPERTY) {
-	//				// replace initializer for variable
-	//				VariableDeclarationFragment vdf = (VariableDeclarationFragment) name.getParent();
-	//				Expression expression = vdf.getInitializer();
-	//				if (expression != null) {
-	//					rewrite.replace(expression, getInversedExpression(rewrite, expression), null);
-	//				}
-	//				// set new name
-	//				rewrite.replace(name, newName, null);
-	//			} else if (name.getParent() instanceof PrefixExpression && ((PrefixExpression) name.getParent()).getOperator() == PrefixExpression.Operator.NOT) {
-	//				rewrite.replace(name.getParent(), newName, null);
-	//			} else {
-	//				PrefixExpression expression = ast.newPrefixExpression();
-	//				expression.setOperator(PrefixExpression.Operator.NOT);
-	//				expression.setOperand(newName);
-	//				rewrite.replace(name, expression, null);
-	//			}
-	//		}
-	//		// add correction proposal
-	//		resultingCollections.add(proposal);
-	//		return true;
-	//	}
-	//
+
+	private boolean getInverseLocalVariableProposals(CodeActionParams params, IInvocationContext context, ASTNode covering, Collection<CUCorrectionProposal> proposals) {
+		if (proposals == null) {
+			return false;
+		}
+
+		CUCorrectionProposal proposal = null;
+		if (this.preferenceManager.getClientPreferences().isAdvancedExtractRefactoringSupported()) {
+			proposal = getInvertVariableCommandProposal(params, context, covering);
+		} else {
+			proposal = getInvertVariableProposal(params, context, covering);
+		}
+
+		if (proposal == null) {
+			return false;
+		}
+
+		proposals.add(proposal);
+		return true;
+	}
+
+	private static CUCorrectionProposal getInvertVariableCommandProposal(CodeActionParams params, IInvocationContext context, ASTNode covering) {

```

Comment:
Given this class is already a super set of functions. These two utilities `getInvertVariableCommandProposal` and `getInvertVariableProposal` look not unnecessary. I would suggest to convert them to inline.

Reply:
Sure